addons/project/static/src/components/subtask_one2many_field/subtask_one2many_field.js

```diff
@@ -0,0 +1,24 @@
+/** @odoo-module */
+
+import { registry } from "@web/core/registry";
+import { X2ManyField } from '@web/views/fields/x2many/x2many_field';
+
+import { SubtaskListRenderer } from './subtask_list_renderer';
+
+export class SubtaskOne2ManyField extends X2ManyField {
+    get Renderer() {
+        return this.viewMode === 'list' ? SubtaskListRenderer : super.Renderer;
+    }
```

Comment:
It is not necessary. 🙂 
It is already done in the template of the X2ManyField component. 🙂 

https://github.com/odoo/odoo/blob/013c0f9c9303e1c8669e01b0125a9d4476939e3f/addons/web/static/src/views/fields/x2many/x2many_field.xml#L35-L36